### ISSUE TITLE
Please tweak default language options for audio and subtitles tracks

### ISSUE THREAD

**Issue by damengilland** [lhc70000/iina#381]
## User Story

I want a way to have iina remember my preferred audio and subtile preferences so that when I'm watching videos, especially marathoning, I don't have to stop at every video to re-set the audio and subtitle tracks.

## Steps to Reproduce / Test

Start a video set up audio and subtitle the way you want, when the video is over and iina jumps to the next video it resets the audio and subtitle tracks.

### iina has the option for audio, but it doesn't work unless it's a perfect match.  If the language listed is "Eng", for example, it will play the file's default.

![screen shot 2017-02-14 at 3 04 56 am](https://cloud.githubusercontent.com/assets/1106778/22922223/649b258e-f262-11e6-8eeb-14ec272fe6c8.png)

### VLC on the other hand has these options, and whatever they are doing behind the scenes keeps the audio and subtitle preferences correct between files.

<img width="682" alt="screen shot 2017-02-14 at 3 04 11 am" src="https://cloud.githubusercontent.com/assets/1106778/22922234/6d9766d4-f262-11e6-9686-16ab4ff83d05.png">

<img width="682" alt="screen shot 2017-02-14 at 3 04 14 am" src="https://cloud.githubusercontent.com/assets/1106778/22922235/6d996ed4-f262-11e6-8f48-d2a31cf4ac95.png">

## Acceptance Criteria / Expected Behavior

I think what would be cool is if you had a dropdown for people to select their default language, and then behind the scenes you did some magic that would know that even though I selected US English, if only English was present to go to English.  (Should just be a cross check against http://www.loc.gov/standards/iso639-2/php/code_list.php ?)  Same functionality for audio and subtitles, but with Subtitles you'd need "default 1" and "default 2" (since you have options of displaying 2), and you need to add a "Forced" option to the languages.

![new mockup 1](https://cloud.githubusercontent.com/assets/1106778/22922657/4af68d42-f264-11e6-9635-b76e90ad4926.png)

## Special Testing Needs

None.

## Dependencies

None.

## Other Notes / Files

None.

**Comment by megadr01d** [comment]
> I think what would be cool is if you had a dropdown for people to select their default language, and then behind the scenes you did some magic that would know that even though I selected US English, if only my locale wasn't selected it could default to the base language and just use the first it found of that.

Should just be a cross check against http://www.loc.gov/standards/iso639-2/php/code_list.php for different spellings -- basically some files show up as Eng vs. English vs. American English, etc. Same functionality for audio and subtitles, but with Subtitles you'd need "default 1" and "default 2" (since you have options of displaying 2), and you need to add a "Forced" option to the languages.

This is complicated because IINA would have to keep this database of synonyms updated. Besides, why a dropdown menu? A text field is the preferred choice for a user to set his own set of language identifiers, separated by commas, eg. `English, Eng, ENG, EN...`?

**Comment by lhc70000** [comment]
mpv tries to reset options as much as possible when switching files. Yes, at least audio and sub language should be excluded.

**Comment by damengilland** [comment]
Dropdown is much easier for users than a text field.  I would argue that average users don't know Eng vs. English or any of the locale codes.

It's really annoying how so many of my files switch to Russian or other languages when I have `Eng` set.  I normally stream to a TV across the room, when marathoning TV shows having to get up and change the language makes this program far inferior to VLC for me at the moment.

**Comment by euf** [comment]
Double-language subtitles are extremely useful, so there should be a way to activate both languages automatically, without having to hit Cmd+S every time.

**Comment by lhc70000** [comment]
I've implemented a better solution for subtitle language in build 15 (internal release). Tell me if you want to have a try.
![8fab17eb-4946-4aeb-90eb-4fa7ab7d07e4](https://cloud.githubusercontent.com/assets/8478049/24022963/47c914bc-0ae4-11e7-861a-f7807a233534.png)

**Comment by euf** [comment]
And what about second language? Is it possible to add there?

**Comment by lhc70000** [comment]
@euf 
Say if you have A and B as sub language, then it should select subtitle in `A > B > others` order. But there's no way to activate two subtitles by setting sub language, if you mean that. Second sub is always disabled initially.

**Comment by euf** [comment]
@lhc70000 
Can it be possible to activate two subtitles initially? 
Maybe I should open an issue on this feature?
Second subtitle language should be customizable the same way.

**Comment by lhc70000** [comment]
No, I said there's no way, at least in mpv. IINA is based on mpv so we can't add arbitrary features that it doesn't have. Sorry for that.

**Comment by damengilland** [comment]
Thanks for adding this.

Only thing I wish is that it would allow for is the concept of "forced" vs just the default language.  Any way to do this?  "ENG - Forced" is typically a language option for subtitles and then you only see subs when the spoken language is something other than English.

**Comment by damengilland** [comment]
Also, small issue... but the window is too small for the Subtitle panel.

Making the entire preference window wider will remove the scroll bars.

![screen shot 2017-04-21 at 2 59 29 pm](https://cloud.githubusercontent.com/assets/1106778/25294179/a9003bfe-26a3-11e7-8a9e-a38235428e7d.png)

**Comment by saagarjha** [comment]
Seems to be the same issue as #486. Could you try going through the steps listed there?

**Comment by Nes-si** [comment]
@lhc70000 you've implemented a nice language choosing, but for subtitles only. Why you didn't add it to audio?

**Comment by megema** [comment]
I mostly watch videos with disabled subtitles (MKVs with built-in subtitles), therefore cannot use this feature, because it forces subtitles to show up. It wouldn't be much of a pain to not to use it, but to be able to download subtitles ("Find online subtitles"), the list have to be populated or I get an error message every time and am able to download only English ones.

Please fix this issue! Populating the "Preferred language" list should not force showing subtitles. That should be controlled by another option.

**Comment by megema** [comment]
Anyone watching this issue?

**Comment by euf** [comment]
@lhc70000 
> No, I said there's no way, at least in mpv. IINA is based on mpv so we can't add arbitrary features that it doesn't have. Sorry for that.

As I can see, mpv has an option to activate two subtitles now.
https://mpv.io/manual/master/#options-secondary-sid
Could you please consider adding a second "preferred language" field to settings?

The double subtitle functionality is a killer feature for me and many other users, but selecting second subtitle manually each time is a huge inconvenience.

**Comment by euf** [comment]
The desired behaviour was explained in #2862
